salt/modules/win_path.py

```diff
@@ -47,15 +47,28 @@ def _normalize_dir(string):
 
 def rehash():
     '''
-    Send a WM_SETTINGCHANGE Broadcast to Windows to refresh the Environment variables
+    Send a WM_SETTINGCHANGE Broadcast to Windows to refresh the Environment
+    variables for new processes.
+
+    .. note::
+        This will only affect new processes that aren't launched by services. To
+        apply changes to the path to services, the host must be restarted. The
+        ``salt-minion``, if running as a service, will not see changes to the
+        environment until the system is restarted. See
+        `MSDN Documentation <https://support.microsoft.com/en-us/help/821761/changes-that-you-make-to-environment-variables-do-not-affect-services>`
 
     CLI Example:
 
     ... code-block:: bash
 
         salt '*' win_path.rehash
     '''
-    return bool(SendMessage(HWND_BROADCAST, WM_SETTINGCHANGE, 0, 'Environment'))
+    broadcast_message = ctypes.create_unicode_buffer('Environment')
+    user32 = ctypes.windll('user32', use_last_error=True)
+    result = user32.SendMessageTimeoutW(HWND_BROADCAST, WM_SETTINGCHANGE, 0,
+                                        broadcast_message, SMTO_ABORTIFHUNG,
+                                        5000, 0)
+    return result == 1
```

Comment:
# https://msdn.microsoft.com/en-us/library/windows/desktop/ms644952(v=vs.85).aspx
    _, res = win32gui.SendMessageTimeout(
        win32con.HWND_BROADCAST, win32con.WM_SETTINGCHANGE, 0, 0,
        win32con.SMTO_ABORTIFHUNG, 5000)

Reply:
The win32gui one didn't work for me. We should move this into a salt.util if it's being used in two places.

Reply:
Move this and reg one into win_functions. 
i.e. win_functions should be common functions shared across the platform.

Reply:
I approve this change as is, or update it,

Reply:
Added `refresh_environment` to `salt.utils.win_functions`.
Pointed the `rehash` function in `win_path.py` to the new salt util
Pointed the `broadcast_change` function in `reg.py` to the new salt util